packages/storybook/src/nimble/table-column/menu-button/table-column-menu-button.stories.ts

```diff
@@ -0,0 +1,187 @@
+import { html, ref } from '@microsoft/fast-element';
+import type { HtmlRenderer, Meta, StoryObj } from '@storybook/html';
+import { withActions } from '@storybook/addon-actions/decorator';
+import { tableTag } from '../../../../../nimble-components/src/table';
+import { tableColumnTextTag } from '../../../../../nimble-components/src/table-column/text';
+import { tableColumnMenuButtonTag } from '../../../../../nimble-components/src/table-column/menu-button';
+import type { MenuButtonColumnToggleEventDetail } from '../../../../../nimble-components/src/table-column/menu-button/types';
+import { Menu, menuTag } from '../../../../../nimble-components/src/menu';
+import { menuItemTag } from '../../../../../nimble-components/src/menu-item';
+import {
+    SharedTableArgs,
+    sharedTableActions,
+    sharedTableArgTypes,
+    sharedTableArgs
+} from '../base/table-column-stories-utils';
+import {
+    apiCategory,
+    createUserSelectedThemeStory
+} from '../../../utilities/storybook';
+
+const simpleData = [
+    {
+        id: 'Ralph Wiggum',
+        firstName: 'Ralph',
+        lastName: 'Wiggum',
+        favoriteColor: 'Rainbow',
+        quote: "I'm in danger!"
+    },
+    {
+        id: 'Milhouse Van Houten',
+        firstName: 'Milhouse',
+        lastName: 'Van Houten',
+        favoriteColor: 'Crimson',
+        quote: "Not only am I not learning, I'm forgetting stuff I used to know!"
+    },
+    {
+        id: 'Ned Flanders',
+        firstName: 'Ned',
+        lastName: 'Flanders',
+        favoriteColor: 'Taupe',
+        quote: 'Hi diddly-ho neighbor!'
+    },
+    {
+        id: 'Maggie Simpson',
+        firstName: 'Maggie',
+        lastName: 'Simpson',
+        favoriteColor: 'Red'
+    }
+] as const;
+
+const metadata: Meta<SharedTableArgs> = {
+    title: 'Components/Table Column: Menu Button',
+    decorators: [withActions<HtmlRenderer>],
+    parameters: {
+        actions: {
+            handles: [
+                ...sharedTableActions,
+                'menu-button-column-toggle',
+                'menu-button-column-beforetoggle'
+            ]
+        }
+    },
+    // prettier-ignore
+    argTypes: {
+        ...sharedTableArgTypes,
+        selectionMode: {
+            table: {
+                disable: true
+            }
+        },
+    },
+    args: {
+        ...sharedTableArgs(simpleData)
+    }
+};
+
+export default metadata;
+
+interface MenuButtonColumnTableArgs extends SharedTableArgs {
+    fieldName: string;
+    menuSlot: string;
+    updateMenuItems: (
+        storyArgs: MenuButtonColumnTableArgs,
+        e: CustomEvent<MenuButtonColumnToggleEventDetail>
+    ) => void;
+    menuRef: Menu;
+    headerContent: string;
+    toggleEvent: never;
+    beforeToggleEvent: never;
+}
+
+export const menuButtonColumn: StoryObj<MenuButtonColumnTableArgs> = {
```

Comment:
It feels weird to me to open the menu button and select a new value but not have the displayed value update. I suspect that most of our client use cases will want to update the display. If that's true, I propose that:
1. we update this code to call `setData` in response to an item being selected
2. we add documentation saying that clients are responsible for doing this. I think that doc would fit either as a short sentence in the overview section or as a longer section under a `## Usage` heading.

Reply:
I've added a `## Usage` heading that mentions that the client needs to handle the `menu-button-column-beforetoggle` event by updating the menu items and handle events on the menu to update the data.